app/controllers/backend/trial_balances_controller.rb

```diff
@@ -26,81 +26,118 @@ def show
       document_nature = Nomen::DocumentNature.find(:trial_balance)
       key = "#{document_nature.name}-#{Time.zone.now.l(format: '%Y-%m-%d-%H:%M:%S')}"
       filename = document_nature.human_name
+      if params[:period] == 'all'
+        params[:started_on] = FinancialYear.all.map { |fy| fy.started_on }.sort.first.to_s
```

Comment:
If you're going to map over all to get a column attribute, use `pluck` instead. You can even combine `order` and `pluck` to avoid the `sort` too.